lxd/instance/drivers/driver_lxc.go

```diff
@@ -3900,6 +3900,15 @@ func (d *lxc) Rename(newName string, applyTemplateTrigger bool) error {
 	// Update lease files.
 	network.UpdateDNSMasqStatic(d.state, "")
 
+	// Reset cloud-init instance-id.
```

Comment:
The comment says what it is doing, but not why? It would be useful for those not familiar with cloud-init's operation to know why we reset the instance ID on rename.

Reply:
Well, it's "because cloud-init wants it done this way" :)
LXD itself really doesn't care.

Reply:
Yeah something along the lines of "Reset cloud-init instance-id to trigger cloud-init refresh on next start"?